### PR TITLE
Fun with flags

### PR DESCRIPTION
**Pull request by luhmirin-s**
* @BurningAXE, added handling of the case you asked in the previous PR
* @alexandr-simprints, added a fix for the issue you found - should have included activity life cycle tracking in the first place, tbh.

### CODE REVIEW

File: feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/ExecutionTracker.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+     * responsibility to handle this case.
+     */
+    val isExecuting = AtomicBoolean(false)
+
+    private var activityCount = 0
+
+    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
+        if (event == Lifecycle.Event.ON_CREATE) {
+            activityCount++
+        } else if (event == Lifecycle.Event.ON_DESTROY) {
```
Comment (BurningAXE) [inline feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/ExecutionTracker.kt:29]: Hmm, what was the use case that this is handling?

File: feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt [hunk 2]
```diff
@@ -83,6 +81,11 @@ internal class OrchestratorFragment : Fragment(R.layout.fragment_orchestrator) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        if (savedInstanceState != null) {
+            // Only updating when explicitly restoring the state
+            orchestratorVm.requestProcessed = savedInstanceState.getBoolean(KEY_REQUEST_PROCESSED)
```
Comment (BurningAXE) [inline feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt:86]: If the flag is moved out of the Fragment, then we don't need to save and restore it anymore?

File: feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/ExecutionTracker.kt (hunk 1 above)
Comment (luhmirin-s) [reply feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/ExecutionTracker.kt:29]: There was a case where you could close an activity before the result callback, and it got stuck with the flag in the "true" state. This ensures that the flag is reset when all activities are closed.

File: feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt (hunk 2 above)
Comment (BurningAXE) [reply feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt:86]: Hmm, I'd argue that this makes it more complex and error-prone. Previously, the request itself (in the args) and the request status were kept inside the Fragment. So, if you just ensure that status is persisted through Fragment restoration, you ensure that the same request won't be handled twice.
Now, a central place keeps the status but if we have two Fragment instances which request is the status referring to?

Comment (luhmirin-s) [reply feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt]: Afaik, the view model is not shared between fragment instances in the case of the parallel flow. Although due to the other fix it should not be an issue anymore.

Comment (luhmirin-s) [reply feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt:84]: After some poking around, this helps when restoring the screen after minimisation. I would rather keep the flag stored in a saved state for now. At least until we figure out the correct way to restore the execution flow.

**Review by meladRaouf — APPROVED** [review]

File: feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt (hunk 2 above)
Comment (BurningAXE) [reply feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/OrchestratorFragment.kt:84]: Oh, right! FSR I was thinking of the Orchestrator as a singleton while yeah, it's an instance connected to the particular Activity/Fragment pair.

**Pull request merged**